magento/magento2
# [2.2] Implement alternative fix for #12285:The option false for mobile device don't work in product view page gallery

### Description
The original fix for #12285 made changes to the way that getVar returns variables from the view.xml file. The original fix changed it so that if the variable content was 'false' or 'true' it would return a boolean value of false or true instead of returning the string value of the boolean as it did before.

Unfortunately this change means that you can no longer tell the difference between the variable existing and being set to false, and the variable not actually existing.

A call to getVar for any random variable name will return false if it doesn't exist, and also false if it happened to exist and was set to 'false'.

This is not good.

This change reverts that change so that the function returns boolean false if the variable name does not exist, and returns the string 'false' if it exists and is set to false. It will always return the string value of the variable contents regardless of the type.

I have reimplemented the fix in a different way, parsing recursively through the Breakpoints array, changing any string 'true' to boolean true and string 'false' to boolean false before parsing into json for sending to browser. This ensures the original #12285 issue remains fixed.

### Fixed Issues (if relevant)
Alternative fix for issue
1. magento/magento2#12285:The option false for mobile device don't work in product view page gallery

### Manual testing scenarios
edit file <theme>/etc/view.xml
add `<var name="allowfullscreen">false</var>` to mobile breakpoints

Example:

```
   <var name="breakpoints">
            <var name="mobile">
                <var name="conditions">
                    <var name="max-width">767px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="navigation">dots</var>
                        <var name="allowfullscreen">false</var>
                    </var>
                </var>
            </var>
        </var>

```
clean cache and open source html
find plugin "mage/gallery/gallery" init options on html page. 

Check that the option "allowfullscreen" value is being output as boolean and not string.

`"breakpoints": {"mobile":{"conditions":{"max-width":"767px"},"options":{"options":{"navigation":"dots","allowfullscreen":false}}}} }
`
### Contribution checklist
